logger/common.go

```diff
@@ -512,7 +512,11 @@ func (l *Logger) sendLogMsgToDest(
 	}
 	err := l.Log(message)
 	if err != nil {
-		return fmt.Errorf("failed to log msg for container %s: %w", l.Info.ContainerName, err)
+		// If we return a non-empty error here, it will cause the goroutine exits. As a result, it won't consume logs from stdout/stderr
+		// and the task container is unable to write logs to stdout/stderr and the application maybe blocked.
+		debug.SendEventsToLog(l.Info.ContainerID,
+			fmt.Sprintf("[Pipe %s] Failed to proxy msg to the log driver : %s", source, err),
+			debug.ERROR, 0)
 	}
 
 	return nil
```

Comment:
As a follow up, we should change the method signature to not even return the error (different PR).

Reply:
ACK